### PR TITLE
[Test] Add gtest to project

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description

Add gtest with a simple example.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by jermainewang** [comment]
@aksnzhy for the review

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,3 +68,16 @@ target_link_libraries(dgl ${DGL_LINKER_LIBS} ${DGL_RUNTIME_LINKER_LIBS})
 
 # Installation rules
 install(TARGETS dgl DESTINATION lib${LIB_SUFFIX})
+
+# Testing
+if(BUILD_TEST)
```
Comment (aksnzhy) [inline CMakeLists.txt]: Can we use `BUILD_CPP_TEST` here?

Comment (VoVAllen) [reply CMakeLists.txt]: Changed to `BUILD_CPP_TEST`

**Comment by zheng-da** [comment]
I think we should invoke the C++ unit tests in CI.

**Pull request merged**